packages/form/core/form-group.ts

```diff
@@ -14,13 +19,25 @@ export class FormGroup {
 	 * @param controls - an array of `FormControl` configuration
 	 * @param name - optional form name
 	 */
-	constructor(controls: ControlConfig[], name = '') {
+	constructor(
+		controls: TypedControlName<FormValues>[],
+		formConfig?: { name?: string; resolver?: ResolvedField }
+	) {
 		const uid = new ShortUniqueId({ length: 9 });
-		this.name = name;
+		this.name = formConfig?.name ?? '';
 		this.id = 'arl-' + uid();
 		this.controls = controls
 			.filter((control) => control.type !== 'submit')
-			.map((control) => new FormControl(control));
+			.map(
+				(control) =>
+					new FormControl({
+						...control,
+						validators:
+							formConfig?.resolver && formConfig.resolver[control.name]
```

Comment:
I would extract this whole logic into a method to make it unit testable. This reminds me, I think we can start unit testing our old typescript classes as well! It was not possible before if I remember correctly. Would be nice if you can check and start a simple unit test, even just for this logic added here. 🙏

Reply:
The logic potentially won't be needed later, so I will prioritize other things first. 

As for testing other classes, I'll do it on separate PR.

Reply:
Ah got it. You can also just open issues for unit tests if others are interested to do that.